docarray/index/backends/hnswlib.py

```diff
@@ -403,7 +406,7 @@ def num_docs(self) -> int:
         """
         Get the number of documents.
         """
-        return self._get_num_docs_sqlite()
+        return self._num_docs
```

Comment:
I think it may be better to handle it here totally: here do:

```python
self._num_docs = self._num_docs or self._get_num_docs_sqlite()
return self._num_docs
```

 And simply in all updates, deletes, index u set self._num_docs to 0.

Reply:
we update the value in the end of index and del, I think this way is more intuitive rather than setting it to 0